chain/pool/txlist.go

```diff
@@ -181,14 +202,35 @@ func (nst *NonceSortedTxs) Get(nonce uint64) (*transaction.Transaction, error) {
 }
 
 func (nst *NonceSortedTxs) GetAllTransactions() []*transaction.Transaction {
+	nst.mu.RLock()
+	defer nst.mu.RUnlock()
+
 	txns := make([]*transaction.Transaction, 0)
-	for _, txnHash := range nst.idx {
-		txns = append(txns, nst.txs[txnHash])
+	if !nst.empty() {
+		for _, txnHash := range nst.idx {
+			txns = append(txns, nst.txs[txnHash])
+		}
 	}
 
 	return txns
 }
 
+func (nst *NonceSortedTxs) GetLatestTxn() (*transaction.Transaction, error) {
+	nst.mu.RLock()
+	defer nst.mu.RUnlock()
+
+	if nst.empty() {
+		return nil, ErrNonceSortedTxsEmpty
+	}
+
+	hash := nst.idx[nst.len()-1]
+	tx, ok := nst.txs[hash]
+	if !ok {
+		panic(fmt.Errorf("%s txList consistency error. Missing idx %s in txs", nst.account.ToHexString(), hash.ToHexString()))
```

Comment:
Should we panic here? We didn't panic on L119

Reply:
nst.idx is a sorted list(by nonce) which txHashe from nst.txs map.keys.
It should never happened when a txHash in nst.idx list but missed in nst.txs map. Otherwise it must a bug.
Let's raise the exception in primary scene if it really happened.

Reply:
Then we should probably panic in both cases right?

Reply:
Done in additional commit.